tests/test_07_l2vpn_return_codes.py

```diff
@@ -0,0 +1,256 @@
+        """
+        Test the return code for creating a SDX L2VPN
+        400: Request does not have a valid JSON or body is incomplete/incorrect
+        -> Body incorrect: port_id
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "Test L2VPN request",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath","vlan": "300"},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "300"}
+            ]
+        }
+        response = requests.patch(f"{api_url}/{self.key}", json=payload)
+        assert response.status_code == 400, response.text
+        
+    @pytest.mark.xfail(reason="return status 500: Internal Server Error")
+    def test_030_edit_l2vpn_with_p2mp_code402(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        402: Request not compatible (For instance, when a L2VPN P2MP is requested but only L2VPN P2P is supported)P2MP
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "VLAN ID 200 at AMPATH, TENET, at SAX", 
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50", "vlan": "200"},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50", "vlan": "200"},
+                {"port_id": "urn:sdx:port:sax.net:Sax01:50","vlan": "200"}
+            ]
+        }
+        response = requests.patch(f"{api_url}/{self.key}", json=payload)
+        assert response.status_code == 402, response.text
+
+    def test_040_edit_l2vpn_not_found_id_code404(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        404: L2VPN Service ID not found
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        key = '11111111-1111-1111-1111-111111111111'
+        response = requests.patch(f"{api_url}/{key}", json=self.payload)
+        assert response.status_code == 404, response.text
+
+    @pytest.mark.xfail(reason="return status 400: PCE error: Can't find a valid vlan breakdown solution")
+    def test_050_edit_l2vpn_conflict_code409(self):
+        """
+        Test the return code for creating a SDX L2VPN
+        409: Conflicts with a different L2VPN
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        # Create a new l2vpn with similar endpoints to the existing one. Only the vlan varies
+        payload = {
+            "name": "Test L2VPN request",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "500"},
+                {"port_id": "urn:sdx:port:sax.net:Sax01:50","vlan": "500"}
+            ]
+        }
+        response = requests.post(api_url, json=payload)
+        assert response.status_code == 201, response.text
+
+        # edit the first l2pvn to match the newly created one
+        self.payload['endpoints'][0]['vlan'] = "500"
+        self.payload['endpoints'][0]['vlan'] = "500"
```

Comment:
Lines are duplicated

Reply:
That's, my mistake, the second line is `self.payload['endpoints'][1]['vlan'] = "500"`, thanks.